diff --git src/linkedlist-lock/coupling.c src/linkedlist-lock/coupling.c
index debf3ae..5251e32 100644
--- src/linkedlist-lock/coupling.c
+++ src/linkedlist-lock/coupling.c
@@ -58,6 +58,7 @@ int lockc_delete(intset_l_t *set, val_t val) {
 
 int lockc_find(intset_l_t *set, val_t val) {
 	node_l_t *curr, *next; 
+	int found;
 	
 	LOCK(&set->head->lock);
 	curr = set->head;
@@ -69,10 +70,11 @@ int lockc_find(intset_l_t *set, val_t val) {
 		curr = next;
 		LOCK(&next->next->lock);
 		next = curr->next;
-	}
+	}	
+	found = (val == next->val);
 	UNLOCK(&curr->lock);
 	UNLOCK(&next->lock);
-	return (next->val == val);
+	return found;
 }
 
 int lockc_insert(intset_l_t *set, val_t val) {
@@ -96,7 +98,6 @@ int lockc_insert(intset_l_t *set, val_t val) {
 	if (!found) {
 		newnode =  new_node_l(val, next, 0);
 		curr->next = newnode;
-		//UNLOCK(&newnode->lock);
 	}
 	UNLOCK(&curr->lock);
 	UNLOCK(&next->lock);

